Flacky test: test_processes_crash_handling_after_executor_gc

Observed on `test_loky macos-py310`:

```
2022-02-04T14:50:14.7293930Z =================================== FAILURES ===================================
2022-02-04T14:50:14.7294690Z _ TestsProcessPoolLokyShutdown.test_processes_crash_handling_after_executor_gc _
2022-02-04T14:50:14.7295640Z 
2022-02-04T14:50:14.7296220Z self = <tests.test_process_executor_loky.TestsProcessPoolLokyShutdown object at 0x116cd9d50>
2022-02-04T14:50:14.7296650Z 
2022-02-04T14:50:14.7297130Z     def test_processes_crash_handling_after_executor_gc(self):
2022-02-04T14:50:14.7297710Z         # Start 5 easy jobs on 5 workers
2022-02-04T14:50:14.7298260Z         results = self.executor.map(sleep_and_return,
2022-02-04T14:50:14.7298790Z                                     [0.01] * 5, range(5))
2022-02-04T14:50:14.7299220Z     
2022-02-04T14:50:14.7299780Z         # Enqueue a job that will trigger a crash of one of the workers.
2022-02-04T14:50:14.7301170Z         # Make sure this crash does not happen before the non-failing jobs
2022-02-04T14:50:14.7301930Z         # have returned their results by using and multiprocessing Event
2022-02-04T14:50:14.7302520Z         # instance
2022-02-04T14:50:14.7302990Z     
2022-02-04T14:50:14.7303550Z         crash_result = self.executor.submit(self._wait_and_crash)
2022-02-04T14:50:14.7304240Z         assert len(self.executor._processes) == self.worker_count
2022-02-04T14:50:14.7304870Z         processes = self.executor._processes
2022-02-04T14:50:14.7305480Z         executor_flags = self.executor._flags
2022-02-04T14:50:14.7306000Z     
2022-02-04T14:50:14.7306520Z         # The following should trigger the GC and therefore shutdown of
2022-02-04T14:50:14.7307100Z         # workers. However the shutdown wait for all the pending jobs to
2022-02-04T14:50:14.7307720Z         # complete first.
2022-02-04T14:50:14.7308290Z         executor_reference = weakref.ref(self.executor)
2022-02-04T14:50:14.7308820Z         self.executor = None
2022-02-04T14:50:14.7309230Z     
2022-02-04T14:50:14.7309690Z         if IS_PYPY:
2022-02-04T14:50:14.7310290Z             # Object deletion and garbage collection can be delayed under PyPy.
2022-02-04T14:50:14.7311510Z             time.sleep(1.)
2022-02-04T14:50:14.7312010Z             gc.collect()
2022-02-04T14:50:14.7313350Z     
2022-02-04T14:50:14.7313990Z         # Make sure that there is not other reference to the executor object.
2022-02-04T14:50:14.7314650Z         assert executor_reference() is None
2022-02-04T14:50:14.7315160Z     
2022-02-04T14:50:14.7315870Z         # The remaining jobs should still be processed in the background
2022-02-04T14:50:14.7316530Z         for result, expected in zip(results, range(5)):
2022-02-04T14:50:14.7317210Z             assert result == expected
2022-02-04T14:50:14.7317760Z     
2022-02-04T14:50:14.7318280Z         # Let the crash job know that it can crash now
2022-02-04T14:50:14.7318860Z         _executor_mixin._test_event.set()
2022-02-04T14:50:14.7319380Z     
2022-02-04T14:50:14.7320690Z         # The crashing job should be executed after the non-failing jobs
2022-02-04T14:50:14.7321380Z         # have completed. The crash should be detected.
2022-02-04T14:50:14.7321900Z         match = filter_match(r"SIGSEGV")
2022-02-04T14:50:14.7322430Z         with pytest.raises(TerminatedWorkerError, match=match):
2022-02-04T14:50:14.7322950Z             crash_result.result()
2022-02-04T14:50:14.7323330Z     
2022-02-04T14:50:14.7323740Z         _executor_mixin._test_event.clear()
2022-02-04T14:50:14.7324150Z     
2022-02-04T14:50:14.7324590Z         # The executor flag should have been set at this point.
2022-02-04T14:50:14.7325170Z         assert executor_flags.broken, processes
2022-02-04T14:50:14.7325590Z     
2022-02-04T14:50:14.7326060Z         # Once all pending jobs have completed the executor and threads should
2022-02-04T14:50:14.7326600Z         # terminate automatically.
2022-02-04T14:50:14.7327080Z >       self.check_no_running_workers(patience=2)
2022-02-04T14:50:14.7327390Z 
2022-02-04T14:50:14.7327900Z crash_result = <Future at 0x1173e6a10 state=finished raised TerminatedWorkerError>
2022-02-04T14:50:14.7328570Z executor_flags = <loky.process_executor._ExecutorFlags object at 0x1173e7ca0>
2022-02-04T14:50:14.7329160Z executor_reference = <weakref at 0x117347510; dead>
2022-02-04T14:50:14.7330050Z expected   = 4
2022-02-04T14:50:14.7330930Z match      = 'SIGSEGV'
2022-02-04T14:50:14.7331390Z processes  = {}
2022-02-04T14:50:14.7331770Z result     = 4
2022-02-04T14:50:14.7332330Z results    = <generator object _chain_from_iterable_of_lists at 0x11730e810>
2022-02-04T14:50:14.7333020Z self       = <tests.test_process_executor_loky.TestsProcessPoolLokyShutdown object at 0x116cd9d50>
2022-02-04T14:50:14.7333440Z 
2022-02-04T14:50:14.7333870Z tests/_test_process_executor.py:277: 
2022-02-04T14:50:14.7334540Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-02-04T14:50:14.7334860Z 
2022-02-04T14:50:14.7335850Z cls = <class 'tests.test_process_executor_loky.TestsProcessPoolLokyShutdown'>
2022-02-04T14:50:14.7336580Z patience = 2, sleep_duration = 0.01
2022-02-04T14:50:14.7336860Z 
2022-02-04T14:50:14.7337260Z     @classmethod
2022-02-04T14:50:14.7337780Z     def check_no_running_workers(cls, patience=5, sleep_duration=0.01):
2022-02-04T14:50:14.7338400Z         deadline = time.time() + patience
2022-02-04T14:50:14.7338820Z     
2022-02-04T14:50:14.7339260Z         while time.time() <= deadline:
2022-02-04T14:50:14.7339720Z             time.sleep(sleep_duration)
2022-02-04T14:50:14.7340210Z             p = psutil.Process()
2022-02-04T14:50:14.7340700Z             workers = _running_children_with_cmdline(p)
2022-02-04T14:50:14.7341200Z             if not workers:
2022-02-04T14:50:14.7341590Z                 return
2022-02-04T14:50:14.7341950Z     
2022-02-04T14:50:14.7342410Z         # Patience exhausted: log the remaining workers command line and
2022-02-04T14:50:14.7342900Z         # raise error.
2022-02-04T14:50:14.7343390Z         print("Remaining worker processes command lines:", file=sys.stderr)
2022-02-04T14:50:14.7343970Z         for w, cmdline in workers:
2022-02-04T14:50:14.7344920Z             print(w.pid, w.status(), end='\n', file=sys.stderr)
2022-02-04T14:50:14.7346710Z             print(cmdline, end='\n\n', file=sys.stderr)
2022-02-04T14:50:14.7347310Z >       raise AssertionError(
2022-02-04T14:50:14.7348240Z             f'Expected no more running worker processes but got {len(workers)}'
2022-02-04T14:50:14.7349260Z             f' after waiting {patience:0.3f}s.'
2022-02-04T14:50:14.7349750Z         )
2022-02-04T14:50:14.7350330Z E       AssertionError: Expected no more running worker processes but got 1 after waiting 2.000s.
2022-02-04T14:50:14.7350970Z 
2022-02-04T14:50:14.7351900Z cls        = <class 'tests.test_process_executor_loky.TestsProcessPoolLokyShutdown'>
2022-02-04T14:50:14.7353150Z cmdline    = '/Users/runner/work/1/s/.tox/py310/bin/python -m loky.backend.popen_loky_posix --process-name LokyProcess-356 --pipe 249'
2022-02-04T14:50:14.7353950Z deadline   = 1643985856.845193
2022-02-04T14:50:14.7354990Z p          = psutil.Process(pid=2269, name='python3.10', status='running', started='14:40:59')
2022-02-04T14:50:14.7355800Z patience   = 2
2022-02-04T14:50:14.7356310Z sleep_duration = 0.01
2022-02-04T14:50:14.7357340Z w          = psutil.Process(pid=6974, name='python3.10', status='running', started='14:44:13')
2022-02-04T14:50:14.7359020Z workers    = [(psutil.Process(pid=6974, name='python3.10', status='running', started='14:44:13'), '/Users/runner/work/1/s/.tox/py310/bin/python -m loky.backend.popen_loky_posix --process-name LokyProcess-356 --pipe 249')]
2022-02-04T14:50:14.7359810Z 
2022-02-04T14:50:14.7360280Z tests/_executor_mixin.py:184: AssertionError
2022-02-04T14:50:14.7361270Z ---------------------------- Captured stderr setup -----------------------------
2022-02-04T14:50:14.7362040Z [DEBUG:MainProcess:MainThread] created semlock with handle 136 and name "None"
2022-02-04T14:50:14.7362770Z [DEBUG:MainProcess:MainThread] created semlock with handle 141 and name "None"
2022-02-04T14:50:14.7363450Z [DEBUG:MainProcess:MainThread] created semlock with handle 142 and name "None"
2022-02-04T14:50:14.7364150Z [DEBUG:MainProcess:MainThread] created semlock with handle 191 and name "None"
2022-02-04T14:50:14.7364830Z [DEBUG:MainProcess:MainThread] created semlock with handle 194 and name "None"
2022-02-04T14:50:14.7365850Z [DEBUG:MainProcess:MainThread] created semlock with handle 195 and name "None"
2022-02-04T14:50:14.7366500Z [DEBUG:MainProcess:MainThread] ProcessPoolExecutor is setup
2022-02-04T14:50:14.7367190Z [DEBUG:MainProcess:MainThread] created semlock with handle 246 and name "None"
2022-02-04T14:50:14.7367840Z [DEBUG:MainProcess:MainThread] launched python with pid 6974 and cmd:
2022-02-04T14:50:14.7369120Z ['/Users/runner/work/1/s/.tox/py310/bin/python', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-356', '--pipe', '249']
2022-02-04T14:50:14.7369990Z [DEBUG:MainProcess:MainThread] created semlock with handle 248 and name "None"
2022-02-04T14:50:14.7370700Z [DEBUG:MainProcess:MainThread] launched python with pid 6975 and cmd:
2022-02-04T14:50:14.7371920Z ['/Users/runner/work/1/s/.tox/py310/bin/python', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-357', '--pipe', '251']
2022-02-04T14:50:14.7372820Z [DEBUG:MainProcess:MainThread] created semlock with handle 250 and name "None"
2022-02-04T14:50:14.7373490Z [DEBUG:MainProcess:MainThread] launched python with pid 6976 and cmd:
2022-02-04T14:50:14.7374740Z ['/Users/runner/work/1/s/.tox/py310/bin/python', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-358', '--pipe', '253']
2022-02-04T14:50:14.7375640Z [DEBUG:MainProcess:MainThread] created semlock with handle 252 and name "None"
2022-02-04T14:50:14.7376300Z [DEBUG:MainProcess:MainThread] launched python with pid 6977 and cmd:
2022-02-04T14:50:14.7377540Z ['/Users/runner/work/1/s/.tox/py310/bin/python', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-359', '--pipe', '255']
2022-02-04T14:50:14.7378440Z [DEBUG:MainProcess:MainThread] created semlock with handle 254 and name "None"
2022-02-04T14:50:14.7379120Z [DEBUG:MainProcess:MainThread] launched python with pid 6978 and cmd:
2022-02-04T14:50:14.7380750Z ['/Users/runner/work/1/s/.tox/py310/bin/python', '-m', 'loky.backend.popen_loky_posix', '--process-name', 'LokyProcess-360', '--pipe', '257']
2022-02-04T14:50:14.7383030Z [DEBUG:MainProcess:MainThread] Adjust process count : {6974: <LokyProcess name='LokyProcess-356' pid=6974 parent=2269 started>, 6975: <LokyProcess name='LokyProcess-357' pid=6975 parent=2269 started>, 6976: <LokyProcess name='LokyProcess-358' pid=6976 parent=2269 started>, 6977: <LokyProcess name='LokyProcess-359' pid=6977 parent=2269 started>, 6978: <LokyProcess name='LokyProcess-360' pid=6978 parent=2269 started>}
2022-02-04T14:50:14.7384530Z [DEBUG:MainProcess:MainThread] _start_executor_manager_thread called
2022-02-04T14:50:14.7385290Z [DEBUG:MainProcess:ExecutorManagerThread] Queue._start_thread()
2022-02-04T14:50:14.7386000Z [DEBUG:MainProcess:ExecutorManagerThread] doing self._thread.start()
2022-02-04T14:50:14.7386650Z [DEBUG:MainProcess:QueueFeederThread] starting thread to feed data to pipe
2022-02-04T14:50:14.7388520Z [DEBUG:MainProcess:ExecutorManagerThread] ... done self._thread.start()
2022-02-04T14:50:14.7390300Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 141 and name "/loky-2269-uz9l7qrb"
2022-02-04T14:50:14.7391670Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 142 and name "/loky-2269-b6hs5_mg"
2022-02-04T14:50:14.7393760Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 191 and name "/loky-2269-b_wr04mb"
2022-02-04T14:50:14.7395080Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 194 and name "/loky-2269-x2e9qaw_"
2022-02-04T14:50:14.7396350Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 195 and name "/loky-2269-g1_jw5hc"
2022-02-04T14:50:14.7397580Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 141 and name "/loky-2269-uz9l7qrb"
2022-02-04T14:50:14.7398910Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 142 and name "/loky-2269-b6hs5_mg"
2022-02-04T14:50:14.7400240Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 191 and name "/loky-2269-b_wr04mb"
2022-02-04T14:50:14.7401780Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 194 and name "/loky-2269-x2e9qaw_"
2022-02-04T14:50:14.7403040Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 141 and name "/loky-2269-uz9l7qrb"
2022-02-04T14:50:14.7404310Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 142 and name "/loky-2269-b6hs5_mg"
2022-02-04T14:50:14.7405570Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 191 and name "/loky-2269-b_wr04mb"
2022-02-04T14:50:14.7406830Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 194 and name "/loky-2269-x2e9qaw_"
2022-02-04T14:50:14.7408090Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 195 and name "/loky-2269-g1_jw5hc"
2022-02-04T14:50:14.7409330Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 141 and name "/loky-2269-uz9l7qrb"
2022-02-04T14:50:14.7410600Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 142 and name "/loky-2269-b6hs5_mg"
2022-02-04T14:50:14.7411870Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 191 and name "/loky-2269-b_wr04mb"
2022-02-04T14:50:14.7413080Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 194 and name "/loky-2269-x2e9qaw_"
2022-02-04T14:50:14.7414330Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 195 and name "/loky-2269-g1_jw5hc"
2022-02-04T14:50:14.7415580Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 195 and name "/loky-2269-g1_jw5hc"
2022-02-04T14:50:14.7416830Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 141 and name "/loky-2269-uz9l7qrb"
2022-02-04T14:50:14.7418090Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 142 and name "/loky-2269-b6hs5_mg"
2022-02-04T14:50:14.7419300Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 191 and name "/loky-2269-b_wr04mb"
2022-02-04T14:50:14.7420960Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 194 and name "/loky-2269-x2e9qaw_"
2022-02-04T14:50:14.7422210Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 195 and name "/loky-2269-g1_jw5hc"
2022-02-04T14:50:14.7423620Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 131 and name "/loky-2269-43kwnqk9"
2022-02-04T14:50:14.7424960Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 132 and name "/loky-2269-y8x733sp"
2022-02-04T14:50:14.7426220Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 133 and name "/loky-2269-jo55ksuv"
2022-02-04T14:50:14.7427470Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 134 and name "/loky-2269-pchxhy87"
2022-02-04T14:50:14.7428720Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 135 and name "/loky-2269-buuy4fuu"
2022-02-04T14:50:14.7429970Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 136 and name "/loky-2269-yvvaqqf_"
2022-02-04T14:50:14.7431240Z [DEBUG:LokyProcess-359:MainThread] recreated blocker with handle 252 and name "/loky-2269-cw9jcixd"
2022-02-04T14:50:14.7432370Z [INFO:LokyProcess-359:MainThread] child process calling self.run()
2022-02-04T14:50:14.7433480Z [DEBUG:LokyProcess-359:MainThread] Worker started with timeout=None
2022-02-04T14:50:14.7434590Z ----------------------------- Captured stderr call -----------------------------
2022-02-04T14:50:14.7435560Z [DEBUG:MainProcess:MainThread] Executor collected: triggering callback for QueueManager wakeup
2022-02-04T14:50:14.7436750Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 131 and name "/loky-2269-43kwnqk9"
2022-02-04T14:50:14.7438020Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 132 and name "/loky-2269-y8x733sp"
2022-02-04T14:50:14.7439270Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 133 and name "/loky-2269-jo55ksuv"
2022-02-04T14:50:14.7440520Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 134 and name "/loky-2269-pchxhy87"
2022-02-04T14:50:14.7442020Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 135 and name "/loky-2269-buuy4fuu"
2022-02-04T14:50:14.7443290Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 136 and name "/loky-2269-yvvaqqf_"
2022-02-04T14:50:14.7444550Z [DEBUG:LokyProcess-358:MainThread] recreated blocker with handle 250 and name "/loky-2269-2_21hpq5"
2022-02-04T14:50:14.7445810Z [INFO:LokyProcess-358:MainThread] child process calling self.run()
2022-02-04T14:50:14.7446920Z [DEBUG:LokyProcess-358:MainThread] Worker started with timeout=None
2022-02-04T14:50:14.7447970Z [DEBUG:LokyProcess-359:MainThread] psutil return memory size: 26742784
2022-02-04T14:50:14.7449180Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 131 and name "/loky-2269-43kwnqk9"
2022-02-04T14:50:14.7450400Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 132 and name "/loky-2269-y8x733sp"
2022-02-04T14:50:14.7451660Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 133 and name "/loky-2269-jo55ksuv"
2022-02-04T14:50:14.7452910Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 134 and name "/loky-2269-pchxhy87"
2022-02-04T14:50:14.7454170Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 135 and name "/loky-2269-buuy4fuu"
2022-02-04T14:50:14.7455430Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 136 and name "/loky-2269-yvvaqqf_"
2022-02-04T14:50:14.7456650Z [DEBUG:LokyProcess-357:MainThread] recreated blocker with handle 248 and name "/loky-2269-x0fxf87q"
2022-02-04T14:50:14.7457830Z [INFO:LokyProcess-357:MainThread] child process calling self.run()
2022-02-04T14:50:14.7458910Z [DEBUG:LokyProcess-357:MainThread] Worker started with timeout=None
2022-02-04T14:50:14.7460120Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 131 and name "/loky-2269-43kwnqk9"
2022-02-04T14:50:14.7461330Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 132 and name "/loky-2269-y8x733sp"
2022-02-04T14:50:14.7462940Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 133 and name "/loky-2269-jo55ksuv"
2022-02-04T14:50:14.7464200Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 134 and name "/loky-2269-pchxhy87"
2022-02-04T14:50:14.7465450Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 135 and name "/loky-2269-buuy4fuu"
2022-02-04T14:50:14.7466700Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 136 and name "/loky-2269-yvvaqqf_"
2022-02-04T14:50:14.7467960Z [DEBUG:LokyProcess-360:MainThread] recreated blocker with handle 254 and name "/loky-2269-ayycr2se"
2022-02-04T14:50:14.7469140Z [INFO:LokyProcess-360:MainThread] child process calling self.run()
2022-02-04T14:50:14.7470240Z [DEBUG:LokyProcess-360:MainThread] Worker started with timeout=None
2022-02-04T14:50:14.7471420Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 131 and name "/loky-2269-43kwnqk9"
2022-02-04T14:50:14.7472690Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 132 and name "/loky-2269-y8x733sp"
2022-02-04T14:50:14.7473950Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 133 and name "/loky-2269-jo55ksuv"
2022-02-04T14:50:14.7475280Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 134 and name "/loky-2269-pchxhy87"
2022-02-04T14:50:14.7476540Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 135 and name "/loky-2269-buuy4fuu"
2022-02-04T14:50:14.7477780Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 136 and name "/loky-2269-yvvaqqf_"
2022-02-04T14:50:14.7479040Z [DEBUG:LokyProcess-356:MainThread] recreated blocker with handle 246 and name "/loky-2269-8op2sav9"
2022-02-04T14:50:14.7480210Z [INFO:LokyProcess-356:MainThread] child process calling self.run()
2022-02-04T14:50:14.7481320Z [DEBUG:LokyProcess-356:MainThread] Worker started with timeout=None
2022-02-04T14:50:14.7482390Z [DEBUG:LokyProcess-360:MainThread] psutil return memory size: 45117440
2022-02-04T14:50:14.7483460Z [DEBUG:LokyProcess-357:MainThread] psutil return memory size: 45121536
2022-02-04T14:50:14.7484780Z [DEBUG:LokyProcess-358:MainThread] psutil return memory size: 45109248
2022-02-04T14:50:14.7486200Z [DEBUG:MainProcess:ExecutorManagerThread] A worker unexpectedly terminated. Workers that might have caused the breakage: {'LokyProcess-359': -11}
2022-02-04T14:50:14.7487630Z [DEBUG:MainProcess:ExecutorManagerThread] terminate process LokyProcess-360, reason: broken executor
2022-02-04T14:50:14.7488780Z [DEBUG:LokyProcess-356:MainThread] psutil return memory size: 45117440
2022-02-04T14:50:14.7489930Z [DEBUG:MainProcess:ExecutorManagerThread] terminate process LokyProcess-359, reason: broken executor
2022-02-04T14:50:14.7491140Z [DEBUG:MainProcess:ExecutorManagerThread] terminate process LokyProcess-358, reason: broken executor
2022-02-04T14:50:14.7492370Z [DEBUG:MainProcess:ExecutorManagerThread] terminate process LokyProcess-357, reason: broken executor
2022-02-04T14:50:14.7495810Z Remaining worker processes command lines:[DEBUG:MainProcess:ExecutorManagerThread] terminate process LokyProcess-356, reason: broken executor
2022-02-04T14:50:14.7496450Z 
2022-02-04T14:50:14.7496890Z 6974 running
2022-02-04T14:50:14.7498110Z /Users/runner/work/1/s/.tox/py310/bin/python -m loky.backend.popen_loky_posix --process-name LokyProcess-356 --pipe 249
2022-02-04T14:50:14.7498560Z 
2022-02-04T14:50:14.7499440Z --------------------------- Captured stderr teardown ---------------------------
2022-02-04T14:50:14.7500130Z [DEBUG:MainProcess:ExecutorManagerThread] closing call_queue
2022-02-04T14:50:14.7500750Z [DEBUG:MainProcess:ExecutorManagerThread] telling queue thread to quit
2022-02-04T14:50:14.7501350Z [DEBUG:MainProcess:ExecutorManagerThread] Queue.join_thread()
2022-02-04T14:50:14.7502350Z [DEBUG:MainProcess:QueueFeederThread] feeder thread got sentinel -- exiting
2022-02-04T14:50:14.7503040Z [DEBUG:MainProcess:ExecutorManagerThread] closing result_queue
2022-02-04T14:50:14.7503880Z [DEBUG:MainProcess:ExecutorManagerThread] closing thread_wakeup
2022-02-04T14:50:14.7504470Z [DEBUG:MainProcess:ExecutorManagerThread] joining processes
2022-02-04T14:50:14.7505150Z [DEBUG:MainProcess:ExecutorManagerThread] executor management thread clean shutdown of worker processes: []
```

Observed here: https://dev.azure.com/joblib/loky/_build/results?buildId=1617&view=logs&j=6159417f-45b9-5b78-cea4-b4926133a526&t=372c4ac8-e768-5a9a-4681-856616da10bb

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -215,7 +215,7 @@ class ExecutorShutdownTest:
             assert result == expected
 
         # Once all pending jobs have completed the executor and threads should
-        # terminate automatically. Note that the effictive time for Python
+        # terminate automatically. Note that the effective time for a Python
         # process to completely shutdown can vary a lot especially on loaded CI
         # machines with and the atexit callbacks that writes test coverage data
         # to disk. Let's be patient.
@@ -803,7 +803,7 @@ class ExecutorTest:
                 pass
 
             # Check that all workers shutdown (via timeout) when waiting a bit:
-            # note that the effictive time for Python process to completely
+            # note that the effective time for a Python process to completely
             # shutdown can vary a lot especially on loaded CI machines with and
             # the atexit callbacks that writes test coverage data to disk.
             # Let's be patient.

--- a/tests/_test_process_executor.py
+++ b/tests/_test_process_executor.py
@@ -215,8 +215,11 @@ class ExecutorShutdownTest:
             assert result == expected
 
         # Once all pending jobs have completed the executor and threads should
-        # terminate automatically.
-        self.check_no_running_workers(patience=2)
+        # terminate automatically. Note that the effictive time for Python
+        # process to completely shutdown can vary a lot especially on loaded CI
+        # machines with and the atexit callbacks that writes test coverage data
+        # to disk. Let's be patient.
+        self.check_no_running_workers(patience=5)
         assert executor_flags.shutdown, processes
         assert not executor_flags.broken, processes
 
@@ -272,9 +275,10 @@ class ExecutorShutdownTest:
         # The executor flag should have been set at this point.
         assert executor_flags.broken, processes
 
-        # Once all pending jobs have completed the executor and threads should
-        # terminate automatically.
-        self.check_no_running_workers(patience=2)
+        # Since the executor is broken, all workers should be SIGKILLed on
+        # POSIX or terminated one Windows. Usually this should be fast but
+        # let's be patient just in case the CI is overloaded.
+        self.check_no_running_workers(patience=5)
 
     def test_context_manager_shutdown(self):
         with self.executor_type(max_workers=5, context=self.context) as e:

--- a/tests/test_synchronize.py
+++ b/tests/test_synchronize.py
@@ -48,7 +48,7 @@ def assert_sem_value_equal(sem, value):
 
 
 def assert_timing_almost_equal(t1, t2=0):
-    assert abs(t1 - t2) < 1e-1
+    assert abs(t1 - t2) < 2e-1
 
 
 class TestLock:

